src/contourbuilder.rs

```diff
@@ -91,11 +91,11 @@ impl ContourBuilder {
                     let v1 = values[ix];
                     if x > 0.0 && x < (dx as Float) && (xt as Float - x).abs() < Float::EPSILON {
                         v0 = values[yt * dx + xt - 1];
-                        point.x = x + (value - v0) / (v1 - v0) - 0.5;
+                        point.x = x + num_traits::cast::<V, Float>( (value - v0) / (v1 - v0) - num_traits::cast(0.5).unwrap()).unwrap();
```

Comment:
This was a temporary hack, I need to figure out a better way to do this. The cast might fail if V = f64 or u128 and Float = f32.